hack/rpm-prepare.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+echo "Reading MGC_VERSION, MCG_RELEASE and GIT_COMMIT env, if they are set"
+# Change version as needed. In most cases MGC_RELEASE would not be touched unless
+# we want to do a re-lease of same version as we are not backporting.
+export MGC_VERSION=${MGC_VERSION:=0.0.1}
+export MGC_RELEASE=${MGC_RELEASE:=1}
+
+export GIT_COMMIT=${GIT_COMMIT:=$(git rev-parse --short HEAD 2>/dev/null)}
+export MGC_RPM_VERSION=${MGC_VERSION//-}
+
+# Golang version variables, if you are bumping this, please contact redhat maintainers to ensure that internal
+# build systems can handle these versions
+export GOLANG_VERSION=${GOLANG_VERSION:-1.15}
+export GOLANG_VERSION_NODOT=${GOLANG_VERSION_NODOT:-115}
+
+# Print env for verification
+echo "Printing envs for verification"
+echo "MGC_VERSION=$MGC_VERSION"
+echo "MGC_RELEASE=$MGC_RELEASE"
+echo "GIT_COMMIT=$GIT_COMMIT"
+echo "MGC_RPM_VERSION=$MGC_RPM_VERSION"
+echo "GOLANG_VERSION=$GOLANG_VERSION"
+echo "GOLANG_VERSION_NODOT=$GOLANG_VERSION_NODOT"
+
+OUT_DIR=".rpmbuild"
+DIST_DIR="$(pwd)/dist"
+
+SPEC_DIR="$OUT_DIR/SPECS"
+SOURCES_DIR="$OUT_DIR/SOURCES"
+FINAL_OUT_DIR="$DIST_DIR/rpmbuild"
+
+NAME="must-gather-clean-$MGC_RPM_VERSION-$MGC_RELEASE"
+
+echo "Making release for $NAME, git commit $GIT_COMMIT"
+
+echo "Cleaning up old content"
+rm -rf "$DIST_DIR"
+rm -rf "$FINAL_OUT_DIR"
+
+echo "Configuring output directory $OUT_DIR"
+rm -rf $OUT_DIR
+mkdir -p $SPEC_DIR
+mkdir -p $SOURCES_DIR/$NAME
+mkdir -p "$FINAL_OUT_DIR"
+
+echo "Generating spec file $SPEC_DIR/must-gather-clean.spec"
+envsubst <rpms/must-gather-clean.spec > $SPEC_DIR/must-gather-clean.spec
+
+echo "Generating tarball $SOURCES_DIR/$NAME.tar.gz"
+# Copy code for manipulation
+cp -arf ./* $SOURCES_DIR/$NAME
+pushd $SOURCES_DIR
+pushd $NAME
+# Remove bin if it exists, we dont need it in tarball
+rm -rf ./odo
```

Comment:
Why is `odo` present? Seemed a bit random and more of an artifact of a dirty build environment.

Reply:
Close, these config files and shell scripts were copied over from another project called `openshift-do` or `odo`.